### PR TITLE
Auto publish products

### PR DESCRIPTION
**Pull request by trickreich**

### CODE REVIEW

File: Model/Product/Handler/Message/SynchronizeProductMessageHandler.php [hunk 1]
```diff
@@ -86,8 +93,19 @@ class SynchronizeProductMessageHandler
...
 
+    /**
+     * @var bool
+     */
+    private $autoPublish;
+
+    /**
+     * @var array
+     */
+    private $processedLocales = [];
```
Comment (wachterjohannes) [inline Model/Product/Handler/Message/SynchronizeProductMessageHandler.php]: we should not store this "state" inside the service

File: Model/Product/ProductInformationInterface.php [hunk 2]
```diff
@@ -49,12 +49,17 @@ public function getShortDescription(): string;
 
     public function setShortDescription(string $shortDescription): self;
 
-    public function mapPublishProperties(self $draft): void;
+    public function mapPublishProperties(ProductInformationInterface $draft): void;
```
Comment (wachterjohannes) [inline Model/Product/ProductInformationInterface.php]: why changing this here? self is used more than once - we should change it everywhere or no where :)

Comment (trickreich) [reply Model/Product/ProductInformationInterface.php]: my ide is marking this red

Comment (trickreich) [reply Model/Product/ProductInformationInterface.php]: but I will change it back then

**Pull request merged**